x-pack/platform/plugins/shared/streams_app/public/components/stream_detail_significant_events_view/edit_significant_event_flyout.tsx

```diff
@@ -61,16 +66,34 @@ export const EditSignificantEventFlyout = ({
 
         switch (data.type) {
           case 'single':
-            await upsertQuery(data.query).then(
+            await upsertQuery({
+              ...data.query,
+              feature: data.query.feature
+                ? {
+                    name: data.query.feature.name,
+                    filter: data.query.feature.filter,
+                  }
+                : undefined,
+            }).then(
               () => {
                 notifications.toasts.addSuccess({
                   title: i18n.translate(
                     'xpack.streams.significantEvents.savedSingle.successfullyToastTitle',
                     { defaultMessage: `Saved significant event query successfully` }
                   ),
                 });
+
                 telemetryClient.trackSignificantEventsCreated({
                   count: 1,
+                  count_by_feature_type: {
+                    system: 0,
+                    infrastructure: 0,
+                    technology: 0,
+                    ...(data.query.feature && {
+                      [(data.query.feature as GeneratedSignificantEventQuery['feature'])!.type]: 1,
```

Comment:
can we augment the `SaveData` type with the `feature.type` (or use `Feature` interface) instead of casting here ? We can also remove the initialized type values, or use a map with exhaustive types instead